kustomize/commands/edit/fix/fix.go

```diff
@@ -84,15 +85,19 @@ and the resulting files may not produce the same output when `+"`kustomize build
 We recommend doing this in a clean git repository where the change is easy to undo.`)
 	}
 
+	if err = splitPatches(fSys, m, w); err != nil {
```

Comment:
I should have the conditions we wanted to affect clearer on the issue. We don't want to necessarily touch any `Patches` field that may already exist--where we want this behaviour is in the code that "fixes" the old `patchesStrategicMerge` field by taking its values and inserting them in the newer `patches:`. When we do that, we need to split multi-doc SMP files.

Reply:
OK, I understand the requirement better now and am sure I can adapt what I've done.

Are you thinking this should occur in `FixKustomizationPreMarshalling`?  I'm finding the code organization here a little confusing as to where the "right" place to do this?

Reply:
Specifically here?  See my TODOs
```
                for _, patchStrategicMerge := range k.PatchesStrategicMerge {
			// check this patch is file path select.
			if _, err := fSys.ReadFile(string(patchStrategicMerge)); err == nil {
				// path patch
				// TODO: Handle splitting file patches
				k.Patches = append(k.Patches, Patch{Path: string(patchStrategicMerge)})
			} else {
				// inline string patch
				// TODO: Handle splitting inline patches (assuming this is a thing that can occur)
				k.Patches = append(k.Patches, Patch{Patch: string(patchStrategicMerge)})
			}
		}
		k.PatchesStrategicMerge = nil
```

Reply:
Yes, that looks like the right place to me. Re: whether multi-document inline patches is a thing, I tried it out, and somewhat surprisingly it does work.